### DISCUSSION TITLE
How do I override Bootstrap text with Google Fonts?

### DISCUSSION THREAD

**Discussion by niajohnson26** [twbs/bootstrap#36690] [Q&A] [answered]
How do I override Bootstrap text with Google Fonts? Asking for a friend.

**Comment by mdo** [comment, marked as answer]
Include the font however you like (Google Fonts has many options listed on their site), and then update the `font-family` wherever you need it. Usually it's enough to override that on the `<body>` element.

For example, add this to your HTML inside the `<head>`:

```html
<link rel="preconnect" href="https://fonts.googleapis.com"> 
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
<link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
```

And in your CSS:

```css
body {
  font-family: 'Roboto', sans-serif;
}
```

**Comment by SamuelMiller** [comment]
For customizing heading fonts, I create a css variable for $headings-font-family in my customized Bootstrap with SASS, using the default --bs-font-sans-serif variable as fallback. (https://getbootstrap.com/docs/5.2/customize/sass/#file-structure_

In my custom bootstrap.scss, I add this:

 ```
 // default variable overrides...
$headings-font-family: var(--headings-font-family), var(--bs-font-sans-serif);
@import '../../../node_modules/bootstrap/scss/bootstrap';
```

This allows me to change the default headings font-family easily in my theme's css by defining --headings-font-family. Such as in the following:

theme.css
```
:root {
	--headings-font-family: 'trashhand';
}

@font-face {
	font-family: 'trashhand';
	src: url('../fonts/trashhand/trashhand-webfont.woff2') format('woff2'), url('../fonts/trashhand-webfont.woff') format('woff');
	font-display: swap;
}
```
For privacy reasons, I like to host googles fonts locally. It is also more compliant with European GDPR regulations. Google's privacy policy allows it to collect a large amount of data about users of its services. Serving fonts via Google exposes the users IP address to Google.
